bbq/vm/test/vm_test.go

```diff
@@ -8285,3 +8285,25 @@ func TestGlobalVariables(t *testing.T) {
 		)
 	})
 }
+
+func TestUserInvokesNativeFunction(t *testing.T) {
+
+	t.Parallel()
+
+	result, err := CompileAndInvoke(t,
+		`
+          fun test(): Int? {
+              let opt: UInt8? = 1
+              return opt.map(Int)
```

Comment:
Without the fix in https://github.com/onflow/cadence/pull/3982/files#diff-da1a470674f415c97427d68ce713bd999c2386eaf8ee9bef093c5532ddeae73bR92-R100, this would fail in a very confusing way: The function would result in a doubly-nested optional. While investigating what caused this, I eventually realized that the invocation of the native function also kept running the compiled code after it, which here happens to be a `Transfer` instruction of the return value, which would lead to an additional boxing.